app/controllers/registration/returns/ReturnsController.scala

```diff
@@ -82,10 +90,18 @@ class ReturnsController @Inject()(val keystoreConnector: KeystoreConnector,
     implicit request =>
       implicit profile =>
         ReturnFrequencyForm.form.bindFromRequest.fold(
-          errors => Future.successful(BadRequest(ReturnFrequencyPage(errors))),
+          errors => for {
+            returns <- returnsService.getReturns
+            showAAS <- returnsService.isEligibleForAAS
+            showMonthly = returns.reclaimVatOnMostReturns.contains(true)
+          } yield {
+            BadRequest(returnFrequencyPage(errors, showAAS, showMonthly))
+          },
           success => returnsService.saveFrequency(success) map { _ =>
             if (success == Monthly) {
               Redirect(controllers.routes.BankAccountDetailsController.showHasCompanyBankAccountView())
+            } else if (success == Annual) {
```

Comment:
could this if elif else block be changed to a case match

Reply:
Refactored